examples/open_source_examples/openrouter.py

```diff
@@ -1,38 +1,97 @@
-import instructor
+import os
 import openai
 from pydantic import BaseModel, Field
 from typing import Optional
-from instructor import Mode
+from instructor import Mode, patch, Maybe
+from dotenv import load_dotenv
 
+# Load environment variables
+load_dotenv()
 
-# Load in your API keys
-openrouter_api_key = "OPENROUTER_API_KEY"
+# Extract API key from environment
+openrouter_api_key = os.environ.get("OPENROUTER_API_KEY")
+if not openrouter_api_key:
+    raise ValueError("OPENROUTER_API_KEY is not set in environment variables")
 
-# Set your OpenAI base_url to point to Open router base OpenAI API spec endpoint
+# Base URL for OpenAI
 openrouter_base_url = "https://openrouter.ai/api/v1"
 
-# Create Instructor patched OpenAI client, ensure you're utilizing JSON mode.
-client = instructor.patch(openai.OpenAI(api_key=openrouter_api_key, base_url=openrouter_base_url), mode=Mode.JSON)
+# Pydantic model for user details
 
 
-# Create our simple pydantic UserDetail BaseModel
 class UserDetail(BaseModel):
-    name: str = Field(..., description="Name of the user")
-    age: int
-    occupation: Optional[str] = Field(None, description="Occupation of the user")
+    """Details in the text about the user"""
+    name: str = Field(description="Name extracted from the text")
+    age: int = Field(description="Age extracted from the text")
+    occupation: Optional[str] = Field(
+        default=None, description="Occupation extracted from the text")
 
-# Utilizing teknium/openhermes-2.5-mistral-7b, we need to ensure the system prompt is present
+
+def create_openai_client(api_key: str, base_url: str):
+    """
+    Create and patch OpenAI client.
+    """
+    client = openai.OpenAI(api_key=api_key, base_url=base_url)
+    return patch(client, mode=Mode.JSON)
+
+
+def get_user_details(client, model: str, user_message: dict) -> Maybe(UserDetail):
+    """
+    Extract user details using Open soure model.
+    """
+    maybe_user_detail = Maybe(UserDetail)
```

Comment:
Classes hould be SnakeCase

Reply:
When you say SnakeCase do you mean UserDetail should be User_Detail OR MaybeUserDetail?
I thought this was snake_case, no?